src/internal/_functionName.js

```diff
@@ -0,0 +1,5 @@
+module.exports = function _functionName(f) {
+  // String(x => x) evaluates to "x => x", so the pattern may not match.
+  var match = String(f).match(/^function (\w*)/);
+  return match == null ? '' : match[1];
```

Comment:
Given a choice between one code path and two, I'll choose one unless there's a compelling reason not to do so. Were we to have two code paths—one just for old versions of IE—we'd actually need to run the tests in those old browsers. With one code path, running the tests in Node is sufficient.